### PR TITLE
testing: integration testing checks in CI for linux and OS X

### PR DESCRIPTION
**Pull request by buildbreaker**
Extracting out the integration tests to a separate job from the unit tests.

Also cleaning up after the fix from https://github.com/envoyproxy/envoy-mobile/pull/1401. The changes here revert the os check in the `JniLibrary` and instead relies on bazel hackery only.

Signed-off-by: Alan Chiu <achiu@lyft.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: testing: integration testing checks in CI for linux and OS X 
Risk Level: low
Testing: ci/local
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: .github/workflows/android_integration_tests.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        with:
+          submodules: true
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '8'
+          java-package: jdk
+          architecture: x64
+      - run: |
+          bazel test --test_output=all --build_tests_only //test/java/integration/...
+        name: 'Run Java library integration tests'
```
Comment (buildbreaker) [inline .github/workflows/android_integration_tests.yml]: This is causing: ```ERROR: /github/home/.cache/bazel/_bazel_root/e390684146b86f31340021c5495bb190/external/bazel_tools/tools/android/BUILD:14:27: in :android_sdk attribute of android_tools_defaults_jar rule @bazel_tools//tools/android:android_jar: alias '@bazel_tools//tools/android:sdk' referring to filegroup rule '@androidsdk//:sdk' through '//external:android/sdk' is misplaced here (expected android_sdk)
INFO: Repository org_robolectric_android_all_9_robolectric_4913185_2 instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule jvm_import_external defined at:
  /github/home/.cache/bazel/_bazel_root/e390684146b86f31340021c5495bb190/external/bazel_tools/tools/build_defs/repo/jvm.bzl:217:38: in <toplevel>
ERROR: Analysis of target '//test/java/integration:android_engine_start_test' failed; build aborted: Analysis of target '@bazel_tools//tools/android:android_jar' failed```

**Comment by buildbreaker** [comment]
`java_tests_new` and `kotlin_tests_new` are now named `java_tests_osx` and `kotlin_tests_osx` respectively.

**Review by goaway — COMMENTED** [review]
Some general comments:

The OS isn't called OSX anymore:
s/osx/mac/g

This setup will run integration tests on mac twice.

I suggest simplifying to just have a linux workflow and mac runflow and just run the top-level test target for each.

Jobs can simply be:
java_tests_mac
java_tests_linux
kotlin_tests_mac
kotlin_tests_linux

File: .github/workflows/android_tests.yml [hunk 2]
```diff
@@ -39,5 +41,28 @@ jobs:
...
+        with:
+          submodules: true
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '8'
+          java-package: jdk
+          architecture: x64
+      - run: |
+          bazel test --test_output=all --build_tests_only //test/kotlin/...
+        name: 'Run Kotlin library integration tests'
```
Comment (goaway) [inline .github/workflows/android_tests.yml:68]: Add newline.

Also maybe add brief comment about why java_tests_linux is disabled?

**Review by goaway — APPROVED** [review]

**Pull request merged**